sound/smooth_swing_v2.h

```diff
@@ -8,36 +8,53 @@
 class SmoothSwingV2 : public SaberBasePassThrough {
 public:
   SmoothSwingV2() : SaberBasePassThrough() {}
-
+  
   void Activate(SaberBase* base_font) {
     STDOUT.println("Activating SmoothSwing V2");
     SetDelegate(base_font);
     if (swingl.files_found() != swingh.files_found()) {
       STDOUT.println("Warning, swingl and swingh should have the same number of files.");
     }
     swings_ = std::min<size_t>(swingl.files_found(), swingh.files_found());
+    if (swng.files_found() > 0)
+    {
+      aswings_ = size_t(swng.files_found());
+      STDOUT.print("Accent Swings Detected: ");
+      STDOUT.println(swng.files_found());
+      accent_swings_present = true;
+    }
+    else
+    {
+      accent_swings_present = false;
+      STDOUT.print("Accent Swings NOT Detected: ");
+    }
   }
-
+  
   void Deactivate() {
     SetDelegate(NULL);
     A.Free();
     B.Free();
+    D.Free();
   }
-
+  
   void Swap() {
     Data C = A;
     A = B;
     B = C;
   }
-
+  
   // Should only be done when the volume is near zero.
   void PickRandomSwing() {
     if (!on_) return;
     uint32_t m = millis();
     // No point in picking a new random so soon after picking one.
     if (A.player && m - last_random_ < 1000) return;
     last_random_ = m;
-    int swing = random(swings_);
+    while (swing == randomizer)
```

Comment:
This will hang forever if there is only one swing sound.
Also, we shouldn't pick the sound, we should just play the effect and let it select which one to play.

Reply:
right, one swing would cause an issue there.  What I was trying to do was prevent the same swing being played twice.